ledger-service/http-json/src/main/scala/com/digitalasset/http/domain.scala

```diff
@@ -134,25 +133,34 @@ object domain extends com.daml.fetchcontracts.domain.Aliases {
 
   final case class PartyDetails(identifier: Party, displayName: Option[String], isLocal: Boolean)
 
-  final case class UserRights(canActAs: List[Party], canReadAs: List[Party], isAdmin: Boolean)
+  sealed trait UserRight
```

Comment:
```suggestion
  sealed abstract class UserRight extends Product with Serializable
```

Reply:
I think I've already had this occasion once but I forgot about it, can you tell me why this is necessary?

Reply:
[Here](https://bazaar.launchpad.net/~scompall/stylish-types/trunk/view/head:/src/test/scala/com/nocandysw/stylishtypes/check/ProductIntersectionTypes.scala#L23), and associated `Product` and `Serializable` wart errors just go away. `abstract class` because use the least powerful construct that works.